torch/distributed/distributed_c10d.py

```diff
@@ -1559,14 +1584,14 @@ def broadcast_object_list(object_list, src, group=group.WORLD):
 
     group_backend = get_backend(group)
     is_nccl_backend = group_backend == Backend.NCCL
-    current_device = torch.device("cpu")
+    current_device: Union[int, torch.device] = torch.device("cpu")
     if is_nccl_backend:
         # See note about using torch.cuda.current_device() here in docstring.
         # We cannot simply use my_rank since rank == device is not necessarily
         # true.
         current_device = torch.cuda.current_device()
```

Comment:
This would be a correct fix here
```suggestion
        current_device = torch.device('cuda', torch.cuda.current_device())
```